File: pingYouGou-sellerGoods-interface/src/main/java/com/lemonzuo/sellergoods/service/BrandService.java
package com.lemonzuo.sellergoods.service;

import com.lemonzuo.model.Brand;
import com.lemonzuo.model.PageResult;

import java.util.List;
import java.util.Map;

/**
 * @author LemonZuo
 * @create 2019-08-10 20:47
 * <p>
 * 品牌服务接口
 */
public interface BrandService {
    /**
     * 带条件查询品牌
     * @param brand 查询条件
     * @return
     */
    List<Brand> findAllBrand(Brand brand);

    /**
     * 品牌分页
     *
     * @param pageNum  当前页数
     * @param pageSize 每页记录数
     * @param brand 品牌
     * @return
     */
    PageResult<Brand> findBrandPage(int pageNum, int pageSize, Brand brand);

    /**
     * 新增品牌
     *
     * @param brand
     */
    void addBrand(Brand brand);

    /**
     * 检测品牌信息是否已存在
     *
     * @param brandName
     * @return
     */
    Brand checkBrand(String brandName);

    /**
     * 修改品牌信息
     *
     * @param brand
     * @return
     */
    void updateBrand(Brand brand);

    /**
     * 根据ID删除品牌
     *
     * @param brandId
     */
    void deleteBrand(Long[] brandId);

    /**
     * 查询品牌ID、品牌名
     * @return
     */
    List<Map> findIdAndName();
}

File: pingYouGou-manager-web/src/main/resources/static/js/service/brandService.js
// 定义 Service
app.service("brandService", function ($http, pingYouGouServer) {
    /**
     * 获取分页数据
     * @param page 当前页
     * @param size 每页数据
     * @param name 品牌名
     * @param firstChar 首字母
     */
    this.findPage = function (page, size, name, firstChar) {
        let url;
        if (name == null && firstChar == null) {
            url = `${pingYouGouServer}/brand/findBrandPage.do?pageNum=${page}&pageSize=${size}`;
        } else if (name != null && firstChar == null) {
            url = `${pingYouGouServer}/brand/findBrandPage.do?pageNum=${page}&pageSize=${size}&name=${name}`;
        } else if (name == null && firstChar != null) {
            url = `${pingYouGouServer}/brand/findBrandPage.do?pageNum=${page}&pageSize=${size}&firstChar=${firstChar}`;
        } else {
            url = `${pingYouGouServer}/brand/findBrandPage.do?pageNum=${page}&pageSize=${size}&name=${name}&firstChar=${firstChar}`;
        }
        return $http({
            url: url,
            method: 'GET'
        });
    }

    // 检测输入的品牌是否已经添加
    this.checkBrand = function (brandName) {
        return $http.get(
            `${pingYouGouServer}/brand/checkBrand.do?brandName=${brandName}`
        );
    }

    // 新增品牌
    this.addBrand = function (brandName, brandFirstChar) {
        return $http({
            url: `${pingYouGouServer}/brand/addBrand.do`,
            method: "POST",
            data: {
                name: brandName,
                firstChar: brandFirstChar
            }
        });
    }

    // 删除品牌
    this.deleteBrand = function (selectIds) {
        return $http({
            url: `${pingYouGouServer}/brand/deleteBrand.do?selectIds=${selectIds}`,
            method: "GET"
        });
    }

    // 修改品牌
    this.modifyBrand = function (brandId, brandName, brandFirstChar) {
        return $http({
            url: `${pingYouGouServer}/brand/updateBrand.do`,
            method: "POST",
            data: {
                id: brandId,
                name: brandName,
                firstChar: brandFirstChar
            }
        });
    }

    this.findIdAndName = function () {
        let url = `${pingYouGouServer}/brand/findIdAndName.do`
        return $http({
            url: url,
            method: 'GET'
        })
    }

})